src/main/java/com/atlassian/db/replica/internal/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -37,7 +37,7 @@ public AuroraReplicasDiscoverer(AuroraJdbcUrl readerUrl) {
     }
 
     private List<String> fetchReplicasServerIds(Connection connection) throws SQLException {
-        LinkedList<String> ids = new LinkedList<>();
+        List<String> ids = new LinkedList<>();
```

Comment:
Putting author shoes i would guess that these are the reasons:
-  we don't know up front how many results will be in ResultSet, so it's not possible to initialise ArrayList with correct size
- iterator in calling method is used to walk throughout all collection so random access is not necessary
IMO type of collection doesn't matter here, because we will have up to 15 replicas (scale is pretty low)

Reply:
> we don't know up front how many results will be in ResultSet, so it's not possible to initialise ArrayList with correct size

We know it's up to 15 and the default for ArrayList is 16 :)


> IMO type of collection doesn't matter here

You're right. In most cases, ArrayList is a better choice, so I was just curious why to use LinkedList here.